tests/test_undertone.py

```
@@ -0,0 +1,47 @@
+        "undertone": "warm",
+        "confidence": 1.4,
+    })
+
+    assert result["undertone_confidence"] == 1.0
+    assert result["flattering_metals"] == ["gold"]
+    assert result["flattering_colors_hex"]
+    assert result["colors_to_avoid_hex"]
+
+
+def test_normalize_result_rejects_invalid_undertone():
+    with pytest.raises(ValueError):
```

Ruff (0.15.17)

[warning] `pytest.raises(ValueError)` is too broad, set the `match` parameter or use a more specific exception

(PT011)